lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -24,6 +24,10 @@ class Main < CC::Engine::Analyzers::Base
 
           def process_file(file)
             RubyParser.new.process(File.binread(file), file, TIMEOUT)
+          rescue Timeout::Error
+            warn "TIMEOUT parsing #{file}. Skipping."
+          rescue => e
```

Comment:
> Those rescues were actually necessary to get the code running on MRI, so I'd like to leave them.

What was erroring without them? Everything or only 2.4 code?

> I'll do the followup PR to clean this up and get error handling generally better. Otherwise I'll open an issue pointing here so we have an artifact to track.

I'm cool with this idea if it let's us remove the bare rescues shortly. I would hold off on deploying this image after merging the PR, though. Suppressing general exceptions in engines in production has been problematic before